inc/admin.php

```diff
@@ -509,3 +515,35 @@ function clean_orphaned_posts( $option, $post_type ) {
 
 	return true;
 }
+
+/**
+ * Check if ads.txt file already exists in the server
+ *
+ * @return void
+ */
+function adstxts_check_for_existing_file() {
+	$home_url_parsed = wp_parse_url( home_url() );
+	if ( empty( $home_url_parsed['path'] ) ) {
+		$response   = wp_remote_request( home_url( '/ads.txt' ) );
+		$file_exist = false;
+		if ( ! is_wp_error( $response ) ) {
+			// Get the headers of the response
+			$headers      = wp_remote_retrieve_headers( $response );
+			$content_type = isset( $headers['content-type'] ) ? $headers['content-type'] : '';
+			$file_exist   = strpos( $content_type, 'application/octet-stream' ) !== false;
```

Comment:
While testing, I was getting a false report here (I had an ads.txt file but it wasn't detected). 

How about the adding `header( 'X-Ads-Txt-Generator: https://wordpress.org/plugins/ads-txt/' );` to the output in the main file (both for ads.txt and for app-ads.txt) and then checking for the header:

```suggestion
			// Check the ads.txt generator header.
			$headers    = wp_remote_retrieve_headers( $response );
			$generator  = isset( $headers['X-Ads-Txt-Generator'] ) ? $headers['X-Ads-Txt-Generator'] : '';
			$file_exist = 'https://wordpress.org/plugins/ads-txt/' !== $generator;
```

The generator header would need to be included regardless of whether the post exists or not, so probably right after the get_option() check in each block.